pkg/helpers/helpers.go

```diff
@@ -163,6 +164,247 @@ func ConcatSlices[T any](slices ...[]T) []T {
+	StroageAwsIamUserArn    string   `json:"-"`
+	StorageAwsExternalId    string   `json:"STORAGE_AWS_EXTERNAL_ID"`
+	EncryptionType          string   `json:"ENCRYPTION_TYPE,omitempty"`
+	EncryptionKmsId         string   `json:"ENCRYPTION_KMS_KEY_ID,omitempty"`
+}
+
+type GCSStorageLocation struct {
+	Name                     string   `json:"NAME"`
+	StorageProvider          string   `json:"STORAGE_PROVIDER"`
+	StorageBaseUrl           string   `json:"STORAGE_BASE_URL"`
+	StorageAllowedLocations  []string `json:"-"`
+	StorageGcpServiceAccount string   `json:"-"`
+	EncryptionType           string   `json:"ENCRYPTION_TYPE,omitempty"`
+	EncryptionKmsId          string   `json:"ENCRYPTION_KMS_KEY_ID,omitempty"`
+}
+
+type AzureStorageLocation struct {
+	Name                    string   `json:"NAME"`
+	StorageProvider         string   `json:"STORAGE_PROVIDER"`
+	StorageBaseUrl          string   `json:"STORAGE_BASE_URL"`
+	StorageAllowedLocations []string `json:"-"`
+	AzureTenantId           string   `json:"AZURE_TENANT_ID"`
+	AzureMultiTenantAppName string   `json:"-"`
+	AzureConsentUrl         string   `json:"-"`
+	EncryptionType          string   `json:"-"`
+	EncryptionKmsId         string   `json:"-"`
+}
+
+type StorageLocation struct {
+	Name                 string `json:"NAME"`
+	StorageProvider      string `json:"STORAGE_PROVIDER"`
+	StorageBaseUrl       string `json:"STORAGE_BASE_URL"`
+	StorageAwsRoleArn    string `json:"STORAGE_AWS_ROLE_ARN,omitempty"`
+	StorageAwsExternalId string `json:"STORAGE_AWS_EXTERNAL_ID,omitempty"`
+	EncryptionType       string `json:"ENCRYPTION_TYPE,omitempty"`
+	EncryptionKmsKeyId   string `json:"ENCRYPTION_KMS_KEY_ID,omitempty"`
+	AzureTenantId        string `json:"AZURE_TENANT_ID,omitempty"`
+}
+
+func storageLocationsEqual(s1 StorageLocation, s2 StorageLocation) bool {
+	return s1.Name == s2.Name &&
+		s1.StorageProvider == s2.StorageProvider &&
+		s1.StorageBaseUrl == s2.StorageBaseUrl &&
+		s1.StorageAwsRoleArn == s2.StorageAwsRoleArn &&
+		s1.StorageAwsExternalId == s2.StorageAwsExternalId &&
+		s1.EncryptionType == s2.EncryptionType &&
+		s1.EncryptionKmsKeyId == s2.EncryptionKmsKeyId &&
+		s1.AzureTenantId == s2.AzureTenantId
+}
+
+func validateParsedExternalVolumeDescribed(p ParsedExternalVolumeDescribed) error {
+	if len(p.StorageLocations) == 0 {
+		return fmt.Errorf("No storage locations could be parsed from the external volume.")
+	}
+	if len(p.AllowWrites) == 0 {
+		return fmt.Errorf("The external volume AllowWrites property could not be parsed.")
+	}
+
+	for _, s := range p.StorageLocations {
+		if len(s.Name) == 0 {
+			return fmt.Errorf("A storage location's Name in this volume could not be parsed.")
+		}
+		if len(s.StorageProvider) == 0 {
+			return fmt.Errorf("A storage location's StorageProvider in this volume could not be parsed.")
+		}
+		if len(s.StorageBaseUrl) == 0 {
+			return fmt.Errorf("A storage location's StorageBaseUrl in this volume could not be parsed.")
```

Comment:
I went for `could not be parsed` as I think this would likely be a parsing error - base urls etc. are mandatory on Snowflake side, so it's not possible that it's empty. Happy to change though if you think it'll be clearer with `is empty` 👍